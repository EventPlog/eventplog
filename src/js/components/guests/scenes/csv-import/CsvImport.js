import React from 'react';
import styled from 'styled-components';

const StyleMessengerCheckIn = styled.div`
  > img {
    width: 100%;
  }
`

const MessengerCheckIn = ({ event = {} }) =>
  <StyleMessengerCheckIn>
    <p>Import a guest lists as CSV.</p>
  </StyleMessengerCheckIn>

export default MessengerCheckIn;
