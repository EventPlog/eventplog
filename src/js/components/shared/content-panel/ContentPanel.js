import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  background: white;
  padding: 1rem;
  box-shadow: 1px 2px 4px ${props => lighten(0.45, props.theme.activeLink)};
    
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    border-bottom: 1px solid ${props => lighten(0.2, props.theme.blue)};
    font-size: 2rem;
    font-weight: 900;
    letter-spacing: 0.01rem;
    margin: 0 1rem;
    padding-bottom: 5px;
    
    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    margin: 2rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h5 className="content-header">{ title }</h5>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel