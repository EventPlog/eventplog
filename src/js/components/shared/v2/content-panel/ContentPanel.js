import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { lighten } from 'polished'

// internal
import { media, maxMedia } from 'js/styles/mixins'
import ContentPanelCard from './ContentPanelCard'


const StyledContentPanel = styled.div`
  ${
    media.phone`
      max-width: 100%;
      padding: 0;
      box-shadow: none;
    `
  }
  .content-header {
    margin: 0;
  }

    ${
      maxMedia.tablet`
        margin: 0 0 1rem;
      `
    }
  }
  
  > div.content-body {
    // margin: 2rem 1rem;
    
    ${
      media.tablet`
        margin: 0 0 1rem;
      `
    }
        
    ${
      media.phone`
        margin: 0 0 1rem;
      `
    }
  }
  
`

const ContentPanel = function({
  className = '',
  title,
  style = {},
  bodyStyle = {},
  children
}) {
  return (
    <StyledContentPanel style={style} className={`${className} content-panel`}>
      {title && <h4 className="content-header">{ title }</h4>}
      <div className="content-body" style={bodyStyle}>
        { children }
      </div>
    </StyledContentPanel>
  )
}

ContentPanel.Card = ContentPanelCard

export default ContentPanel
