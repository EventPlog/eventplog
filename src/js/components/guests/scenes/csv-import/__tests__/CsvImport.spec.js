import React from 'react';
import CsvImport from '../CsvImport';
import ContentPanel from 'js/components/shared/content-panel'
import { shallow } from 'enzyme';

describe('CsvImport', () => {
  it('should render correctly', () => {
    const wrapper = shallow( <CsvImport  /> );

    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find(ContentPanel).length).toEqual(1);
  })
});