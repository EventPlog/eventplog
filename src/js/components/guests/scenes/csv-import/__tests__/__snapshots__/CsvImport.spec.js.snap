// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CsvImport should render correctly 1`] = `
<CsvImport__StyleMessengerCheckIn>
  <ContentPanel
    title="Import a guest lists as CSV."
  >
    <HeadsUpInfo />
    <Form
      as="form"
    >
      <Message
        content="Your file upload process have been kicked off successfully. We'll let you know when it's done."
        header="Success!"
        success={true}
      />
      <Message
        content="Something prevented us from successfully uploading your file. Please try again later."
        error={true}
        header="Error!"
      />
      <FormField>
        <label>
          Upload a CSV file
        </label>
        <Input
          accept=".csv"
          id="csv_upload"
          name="csv"
          type="file"
        />
      </FormField>
      <FormField>
        <Button>
          Submit
        </Button>
      </FormField>
    </Form>
  </ContentPanel>
</CsvImport__StyleMessengerCheckIn>
`;
